### ISSUE TITLE
Calendar doesn't show the correct date the first time it opens

### ISSUE THREAD

**Issue by antoniosdv** [TrevorS/bootstrap3-datetimepicker-rails#52]
The widget shows the current date instead of the input value the first time it opens.

I have reviewed the code, and it seems that the problem comes from the following line (it was commented in v4.17.42):

https://github.com/TrevorS/bootstrap3-datetimepicker-rails/blob/v4.17.43/vendor/assets/javascripts/bootstrap-datetimepicker.js#L869

```javascript
//viewDate = date.clone(); // TODO this doesn't work right on first use
```

**Comment by mattrice** [comment]
Agreed - there are [at least a couple](https://github.com/Eonasdan/bootstrap-datetimepicker/issues/1786) [open issues](https://github.com/Eonasdan/bootstrap-datetimepicker/issues/1836) at the original repo, and [a comment on another issue mentioning the same thing](https://github.com/Eonasdan/bootstrap-datetimepicker/issues/1819#issuecomment-269312765).

I solved this for myself by reverting back to the earlier gem version (4.17.37): 

```
gem 'bootstrap3-datetimepicker-rails', '4.17.37'
```

**Comment by antoniosdv** [comment]
Thanks for you help @mattrice 

As you say, that version could be used to avoid the problem by now. Also, I have found a workaround if someone wants to use the last version despite that:

```javascript
$('input.date').datetimepicker().each(function() {
  $(this).val() && $(this).data("DateTimePicker").viewDate($(this).val());
});
```

**Comment by mattrice** [comment]
I appreciate the workaround, @antoniosdv; unfortunately, it did not work in my situation so I am stuck with the older version until this bug is fixed.

**Comment by vietdh85** [comment]
If @TrevorS or someone update to use version 4.17.45 or lastest version (4.17.47), this issue will be fixed.

**Comment by rickychilcott** [comment]
#53 will update to 4.17.47

**Comment by TrevorS** [comment]
Should be fixed with: 2934715911658f0ad00747711774b0f880797dea. Check it out and let me know! Thanks for your patience.

**Comment by antoniosdv** [comment]
Thanks to you @TrevorS !!!
